src/meta/processors/job/AdminJobProcessor.cpp

```diff
@@ -142,5 +111,49 @@ void AdminJobProcessor::process(const cpp2::AdminJobReq& req) {
   onFinished();
 }
 
+nebula::cpp2::ErrorCode AdminJobProcessor::addJobProcess(const cpp2::AdminJobReq& req,
+                                                         cpp2::AdminJobResult& result) {
+  auto cmd = req.get_cmd();
+  auto paras = req.get_paras();
+
+  // All jobs here are the space level, so the last parameter is spaceName.
+  if (paras.empty()) {
+    LOG(INFO) << "Parameter should be not empty";
+    return nebula::cpp2::ErrorCode::E_INVALID_PARM;
+  }
+
+  // Check if space not exists
+  auto spaceName = paras.back();
+  auto spaceRet = getSpaceId(spaceName);
+  if (!nebula::ok(spaceRet)) {
+    auto retCode = nebula::error(spaceRet);
+    LOG(INFO) << "Get space failed, space name: " << spaceName
+              << " error: " << apache::thrift::util::enumNameSafe(retCode);
+    return retCode;
+  }
+
+  // Check if job not exists
+  JobID jId = 0;
+  auto jobExist = jobMgr_->checkJobExist(cmd, paras, jId);
+  if (jobExist) {
+    LOG(INFO) << "Job has already exists: " << jId;
+    result.job_id_ref() = jId;
+    return nebula::cpp2::ErrorCode::SUCCEEDED;
+  }
+
+  folly::SharedMutex::WriteHolder holder(LockUtils::lock());
+  auto jobId = autoIncrementId();
+  if (!nebula::ok(jobId)) {
+    return nebula::error(jobId);
+  }
+
+  JobDescription jobDesc(nebula::value(jobId), cmd, paras);
+  auto errorCode = jobMgr_->addJob(jobDesc, adminClient_);
```

Comment:
The job manager will have one queue for one space, right?

Reply:
Yes, the later pr will do the following work:
Each space has a priority queue.
Jobs in the same space are executed serially according to priority.
Jobs in different spaces are executed in parallel according to the space.